### ISSUE TITLE
Grading in attendance - allow calculation for all sessions past and future.

### ISSUE THREAD

**Issue by zmartinovic** [danmarsden/moodle-mod_attendance#718]
Currently Attendance gives grade for past sessions. It would be helpful if there is an option to calculate grade for all sessions, past and future. This way we can use attendance grade as condition to access other activity. For example, allow students to access some activity only it they have attended all sessions. Currently if there are 10 session, 5 in the past and 5 in the future, and student has attended past 5 sessions he will have a mark of 100% instead of 50%. This way we can use attendance grade to limit access to other activities at least not until all sessions are complete. So it would be nice to have an option to change how if calculates the grade.

**Comment by danmarsden** [comment]
sounds like an interesting use-case - I don't think this is likely to be something I'd work on as a volunteer, but if you have internal development capabilities, pull requests are always welcome - also feel free to reach out privately if you would like to fund the time for a developer to improve the plugin to meet your needs.

**Comment by zishanj** [comment]
This is highly recommended

**Comment by mudmucks2** [comment]
Plus one for this. Reporting on all possible sessions is a useful metric for some course admins

Or even better, a warning flag gets set if it becomes impossible for a student to 'regain' min attendance level with their current attendance level and sessions remaining.

**Comment by TashLD** [comment]
Plus one as well. We would love to use the plugin, but we require to only have students pass when at least 80% of all face-to-face sessions have been attended.

**Comment by danmarsden** [comment]
before everyone gets excited and starts adding their "plus one" comments..... as mentioned above, this isn't something I'd work on as a volunteer(it's probably quite a bit of work)

If you have internal development capabilities, pull requests are always very welcome and feel free to reach out privately if you would like to fund the time for a developer to improve the plugin to meet your needs. (more plus ones aren't going to change the priority I allocate to my time as a volunteer - sorry!)

**Comment by TashLD** [comment]
@zmartinovic A workaround for this could be the following setup. Let's use the example of a repeated (multiple) session with 8 dates. Students need to attend at least 6 out of 8 sessions for the activity to be completed (this can be your unlocking criterion for other activities)

- In the Attendance **Plugin** settings (via Admin menu), **add a new status** to the Default status set, e.g. "NY" for "Not yet attended" (screenshot 1), with 0 points. This will be your default for all students for all sessions, and when students check their status before a session, it will be obvious that the session has not happened yet.

- In the Attendance **activity** settings:

1. Grade to pass: Enter 12 (for the above example, 6 sessions attended/present > 6x2pts = 12pts). (screenshot 2)
2. Completion conditions: Receive a grade - Select **Passing grade** (screenshot 3)

- **Before** you start tracking attendance, **default the attendance for all students to NY** (use the table header in each session to do so for all students). This way all students will start with the explicit points value of 0 (based on the NY status).

Effects:

- On the course page, the _To do_ drop-down will show Receive a grade (tick), Receive a passing grade (cross) (screenshot 4) - This may be confusing for the students, however, when opening the page with the sessions it shows:
- The Failed grade at the top (again, might be confusing), but also the list of all sessions where each is explicitly listed as "Not yet attended" (screenshot 5)
- The activity is **marked as completed** as soon as a student has been marked as having been present at at least 6 sessions; again, this can be your unlocking criterion for other activities. (screenshot 6)

Screenshot 1:
<img width="359" alt="Attendance 1_ Plugin settings - additional status NY" src="https://github.com/user-attachments/assets/fcdfe04b-59e0-4805-a959-4879a56d87c2">

Screenshot 2:
![Attendance 2 _ Activity settings Grade to pass](https://github.com/user-attachments/assets/84443b30-a2a4-41ac-9692-dab57e8de3fe)

Screenshot 3:
![Attendance 3 _ Activity settings Completion conditions](https://github.com/user-attachments/assets/61e2bb87-b6f7-43ca-94bb-8ad0f88e168e)

Screenshot 4 - Student view:
![Attendance 4_ initial drop-down](https://github.com/user-attachments/assets/ff3b5a45-eaca-435e-bd25-d362c09bfea8)

Screenshot 5 - Student view:
<img width="563" alt="Attendance 5 _ Student view BEFORE" src="https://github.com/user-attachments/assets/46cc5699-58cf-478b-8e12-67e2d4d16e46">

Screenshot 6 - Student view:
![Attendance 6_ Student view COMPLETED](https://github.com/user-attachments/assets/0a572c5b-11a5-46eb-8035-9a73bbc3ff47)

**Comment by danmarsden** [comment]
Nice work-around!

**Comment by tafkey** [comment]
The workaround is interesting but breaks the case when the Teachers do not have the capability to change taken attendances (while they have the capability to take / record attendances, they cannot modify existing ones). Which I think is the default for the Teacher role (in contrast to the Editing Teacher role).

**Comment by tafkey** [comment]
There's a "solution" to arrive to the desired behaviour posted in the [Moodle forums back in 2022 by David Genovese](https://moodle.org/mod/forum/discuss.php?d=427527#p1738498).
